src/GraphQL/Resolvers/IFieldResolver.cs

```diff
@@ -27,15 +27,15 @@ public interface IFieldResolver
         /// <summary>
         /// Returns an object, <see langword="null"/>, or a <see cref="Task{TResult}"/> for the specified field. If a task is returned, then this task will be awaited to obtain the actual object.
         /// </summary>
-        object? Resolve(IResolveFieldContext context);
+        ValueTask<object?> ResolveAsync(IResolveFieldContext context);
```

Comment:
And (more important) all resolvers are async now so we can remove ~ 1/2 of all resolver classes. Right?

Reply:
> The main change that saves memory

Well, the main change that saves memory is due to the fact that middleware does not wrap synchronous resolvers with a Task.  Because we use ValueTask through the entire pipeline, there is no wrapping, as you already noticed.

Reply:
We will not have any problems due to `ValueTask` design limitations (await once, etc)?

Reply:
I though there was but later I found that ExecuteNodeAsync returns Task (not `Task<T>`) and so basically if it runs asynchronously then it gets wrapped at that point. Which is necessary for ParallelExecutionStrategy to WaitAll. 

I will review further of course. It may be that performance is better with another approach. My gut feeling is that if there’s ever a true async operation, any allocations necessary are moot, and probably necessary with ValueTask also; it is really the 99% of sync operations we are optimizing for.

Reply:
Bad news - I can not resolve opened conversations. 🤦‍♂️

Reply:
Perhaps you simply cannot here because "sungam3r" opened this conversation?  In any case, just put a 👍 on a conversation and I'll resolve it.  I'm sure we can work through this and hopefully it's only temporary.

Reply:
Probably can't resolve any of my comments though either :-(